src/base/logger.cpp

```diff
@@ -28,20 +28,31 @@
 
 #include "logger.h"
 
+#include <algorithm>
+
 #include <QDateTime>
-#include "base/utils/string.h"
+
+namespace
+{
+    template <typename Q, typename T>
+    Q toQContainer(const T &src, const int offset = 0)
```

Comment:
I would make it less abstract (both the implementation and name):
```c++
template <typename T>
QVector<T> loadFromBuffer(const boost::circular_buffer_space_optimized<T> &in, const size_t offset = 0)
{
   QVector<T> out;
   out.reserve(in.size() - offset);
   std::copy(in.begin() + offset, in.end(), std::back_inserter(out));
   return out;
}
```

Additionally it allows full type deduction:
```c++
return loadFromBuffer(m_messages);
```

Reply:
Done, PR updated, also added another commit that fixes windows build.